automata/fa/gnfa.py

```diff
@@ -126,9 +126,9 @@ def from_nfa(cls, nfa):
 
     def _validate_transition_invalid_symbols(self, start_state, paths):
         """Raise an error if transition symbols are invalid."""
+        check = self.input_symbols.copy() | {'*', '|', '(', ')', '?'}
```

Comment:
@eliotwrobson Do we need to copy() `self.input_symbols` here? I thought union does not mutate the operands.

Reply:
Ah I think you're right, no copy is necessary. Will change.